### PR TITLE
Symfony 4 support

### PR DESCRIPTION
**Pull request by wachterjohannes**
Replaces #199

### CODE REVIEW

**Comment by wachterjohannes** [comment]
@dbu did you have an idea why the tests fail with symfony 4.0? there is a strange error with multiple `cp` commands. did see something like this before? https://travis-ci.org/phpcr/phpcr-shell/jobs/430393461#L661

**Comment by dbu** [comment]
i don't think its cp commands, but some parser having a problem. maybe the yaml parser of symfony? in version 4 its more strict and does not tolerate some crap data that it tolarated before.

can it be that there is some yaml file in this repo that has a "cp" key that is duplicated?

**Comment by wachterjohannes** [comment]
@dbu thats a good hint :) i will check this

**Comment by wachterjohannes** [comment]
@dbu i have found it: https://github.com/phpcr/phpcr-shell/blob/master/src/PHPCR/Shell/Resources/config.dist/alias.yml

the cp alias is duplicated and stands for 2 different things! i have tested it asap in a current installation and `cp` is for copy should i remove the second entry?

**Comment by dbu** [comment]
> the cp alias is duplicated and stands for 2 different things! i have tested it asap in a current installation and cp is for copy should i remove the second entry?

paging @dantleech :-)

**Comment by wachterjohannes** [comment]
@dbu @dantleech as you can see here https://github.com/phpcr/phpcr-shell/pull/200/files#diff-a31654ea4b9244c0b548a8e9c11d0b15L52 the `version:checkpoint` command has two aliases and when we remove one the tests seems to be green

**Comment by dantleech** [comment]
> command has two aliases and when we remove one the tests seems to be green

sounds good to me, I guess somehow before the first one had precedence. `vcp` would be a better alternative (also fine with omitting it)

**Comment by wachterjohannes** [comment]
@dbu @dantleech thanks for your advices - for me the PR is OK i can also re-add the entry with a prefix :) but could you guys take a look at the rest?

File: src/PHPCR/Shell/Resources/config.dist/alias.yml [hunk 1]
```diff
@@ -49,7 +49,6 @@ checkin: version:checkin
 ci: version:checkin
 co: version:checkout
 checkout: version:checkout
-cp: version:checkpoint
 checkpoint: version:checkpoint
```
Comment (dbu) [inline src/PHPCR/Shell/Resources/config.dist/alias.yml]: fine with dropping the cp line as its already copy. having checkpoint as alias seems good enough to me.

File: composer.json [hunk 2]
```diff
@@ -3,25 +3,24 @@
...
-        "phpcr/phpcr-utils": "~1.2",
-        "symfony/finder": "~2.8|~3.0",
-        "symfony/serializer": "~2.8|~3.0",
-        "symfony/yaml": "~2.8|~3.0",
-        "symfony/dependency-injection": "~2.8|~3.0",
-        "symfony/expression-language": "~2.8|~3.0",
+        "phpcr/phpcr-utils": "~1.2", 
+        "symfony/finder": "~2.8|~3.4|~4.0",
+        "symfony/serializer": "~2.8|~3.4|~4.0",
+        "symfony/yaml": "~2.8|~3.4|~4.0",
```
Comment (dantleech) [inline composer.json]: why 3,4?

Comment (wachterjohannes) [reply composer.json]: indeed could be 3.0 i will change that here

**Comment by wachterjohannes** [comment]
@dantleech fixed

File: .travis.yml [hunk 3]
```diff
@@ -1,20 +1,57 @@
...
+      env:
+        - PREFER=""
+        - BEHAT_SUITE=cli
+        - SYMFONY_VERSION=^4.0
 
 before_script:
     - composer self-update
-    - composer require "symfony/symfony" "~2.8" --no-update
-    - composer install
+    - composer require "symfony/symfony" "$SYMFONY_VERSION" --no-update
```
Comment (dbu) [inline .travis.yml]: requiring symfony/symfony will hide dependency problems. the proper solution these days would be something with flex as explained in https://github.com/symfony/lts

but this is no problem introduced with this pull request, so we don't have to fix it in this PR.

File: composer.json (hunk 2 above)
Comment (dbu) [reply composer.json]: sorry, imho this should be 3.4 indeed. all other 3.x versions of symfony are end of life and don't even get security fixes anymore: https://symfony.com/roadmap

**Comment by wachterjohannes** [comment]
@dbu the problem is that travis would have to run 27 builds - should i enhance the matrix? i think in a further PR you have to fix the travis setup at all as you mentioned the symfony require thing :)

**Comment by dbu** [comment]
why 27? that would indeed be too much.

but i count 12: php 5.6 lowest, 7.0, 7.1 and 7.2, each with the 3 suites. where does the rest come from?

**Comment by wachterjohannes** [comment]
@dbu thats done :) i thought about mixing also the prefer-lowest and the versions that would lead into much more builds.

**Comment by dbu** [comment]
thanks a ton!

**Comment by wachterjohannes** [comment]
@dbu thanks for this fast review/merging :) which version will contain this change? and when do you think will you release it?

**Comment by dbu** [comment]
its released as 1.1.1

**Pull request merged**